pkg/kubernetes/secrets/store.go

```diff
@@ -71,16 +76,45 @@ func (s *Store) Resolve(keyName string, keyValue string) (string, error) {
 	s.logger.Debug(fmt.Sprintf("Store.Resolve: ns:%s, keyName:%s, keyValue:%s", s.namespace, keyName, keyValue))
 	key := strings.ToLower(keyValue)
 
-	secret, err := s.clientSet.CoreV1().Secrets(s.namespace).Get(context.Background(), key, metav1.GetOptions{})
+	secret, err := s.clientSet.CoreV1().Secrets(s.namespace).Get(ctx, key, metav1.GetOptions{})
 	if err != nil {
-		return "", err
+		return "", log.Error(err)
 	}
 	if val, ok := secret.Data[SecretDataKey]; !ok {
-		return "", InvalidSecretDataKeyError{msg: fmt.Sprintf(`The secret %s/%s does not have a key named %s. `+
+		return "", log.Error(InvalidSecretDataKeyError{msg: fmt.Sprintf(`The secret %s/%s does not have a key named %s. `+
 			`The kubernetes.secrets plugin requires that the Kubernetes secret is named after the secret referenced in the `+
 			`Porter parameter or credential set, and secret value is stored in a key on the Kubernetes secret named %s`,
-			s.namespace, keyValue, SecretDataKey, SecretDataKey)}
+			s.namespace, keyValue, SecretDataKey, SecretDataKey)})
 	} else {
 		return string(val), nil
 	}
 }
+
+func (s *Store) Create(ctx context.Context, keyName string, keyValue string, value string) error {
+	ctx, log := tracing.StartSpan(ctx)
+	defer log.EndSpan()
+
+	if err := s.connect(); err != nil {
+		return err
+	}
+
+	key := strings.ToLower(keyValue)
+	s.logger.Debug(fmt.Sprintf("Store.Create: ns:%s, keyName:%s, keyValue:%s", s.namespace, keyName, keyValue))
+	if key != SecretSourceType {
+		return log.Error(fmt.Errorf("unsupported secret type: %s. Only %s is supported", keyName, SecretSourceType))
+	}
+
+	byteValue := []byte(value)
+	// TODO: should we try to compress or segment the value so they can work
```

Comment:
Let's remove the comment here and turn it into an issue so that it's easier to find and track. TODO comments like this won't ever get seen again 😀

Reply:
Good point, here you go! https://github.com/getporter/kubernetes-plugins/issues/109